### PR TITLE
[LA-495] Added two more templates and updated templates to use valueFrom

### PR DESCRIPTION
**Pull request by theof-db**
Added two additional templates.
Made all templates use valueFrom so that the python files can use environment variables.
Updated the license to use the correct license.
Cleaned up yaml files to follow same styling.
Removed comments for streamlit-hello-world-app.
Made minor changes to some of the files to clean them up/be more consistent.

Tested all of the templates in e2 dogfood by uploading the files and deploying them. I used an empty SQL warehouse and a llama instruct serving endpoint for testing.

### CODE REVIEW

File: shiny-chatbot-app/app.py [hunk 1]
```diff
@@ -1,40 +1,50 @@
+# Shiny for Python LLM Chat Example with Databricks
 from shiny import App, ui, reactive
 from openai import AsyncOpenAI
 from databricks.sdk import config
+import os
 
+# Defined in `app.yaml`
+_MODEL_NAME = os.getenv("MODEL_ENDPOINT_NAME")
```
Comment (jerrylian-db) [inline shiny-chatbot-app/app.py:8]: Maybe we should have an assertion here like other templates?

File: streamlit-chatbot-app/app.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        access_token=headers.get("X-Forwarded-Access-Token")
+    )
+
+user_info = get_user_info()
+
+model_mapping = {
+    "DBRX Instruct": "databricks-dbrx-instruct",
+    "Meta Llama 3.1 70B Instruct": "databricks-meta-llama-3-1-70b-instruct",
+    "Mixtral-8x7B Instruct": "databricks-mixtral-8x7b-instruct",
+    "Your Own Model": os.getenv("SERVING_ENDPOINT")
```
Comment (jerrylian-db) [inline streamlit-chatbot-app/app.py:26]: Maybe add an assertion for this environment variable and a corresponding error message?

File: shiny-chatbot-app/app.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+command: [
+  "shiny", "run",
+  "--reload",
+  "--host", "0.0.0.0",
+  "--port", "8000",
+  'app.py:app'
+]
+
+env:
+  - name: "MODEL_ENDPOINT_NAME"
```
Comment (jerrylian-db) [inline shiny-chatbot-app/app.yaml:10]: Nit: we could standardize this env variable name to be `SERVING_ENDPOINT`?

File: dash-data-app/app.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+    assert defined('DATABRICKS_WAREHOUSE_ID') and os.getenv('DATABRICKS_WAREHOUSE_ID') != "<your warehouse ID>", \
+        "To use SQL, set DATABRICKS_WAREHOUSE_ID in app.yaml. You can find your SQL Warehouse ID by " \
+        "navigating to SQL Warehouses, clicking on your warehouse, and then looking for the ID next to the Name."
+    assert defined('DATABRICKS_HOST'), \
+        "To run outside of Lakehouse Apps, set the DATABRICKS_HOST environment variable " \
+        "to the name of your Databricks account."
+    assert defined('DATABRICKS_TOKEN') or (defined('DATABRICKS_CLIENT_ID') and defined('DATABRICKS_CLIENT_SECRET')), \
+        "To run outside of Lakehouse Apps, set environment variables for authentication, " \
+        "such as DATABRICKS_TOKEN or DATABRICKS_CLIENT_ID/DATABRICKS_CLIENT_SECRET."
```
Comment (jerrylian-db) [inline dash-data-app/app.py:25]: Suggestion: we can copy these assertions to other templates that use Databricks APIs.

**Review by jerrylian-db — COMMENTED** [review]
Left some feedback/suggestions

File: dash-data-app/app.py (hunk 4 above)
Comment (theof-db) [reply dash-data-app/app.py:25]: I've talked to Andre, and he says these checks should be handled by the sdk, so I've removed them from all of the templates.

**Review by jerrylian-db — APPROVED** [review]

**Pull request merged**